### ISSUE TITLE
*Need help* Traefik SSL generation

### ISSUE THREAD

**Issue by castrolem** [AdrienPoupa/docker-compose-nas#102]
Hello 👋 , I followed the steps to add my A DNS record into Cloudflare yesterday and I ran the lego validation today, which returned
```
2024/11/29 11:13:38 [INFO] [nas.mydomainname.here] acme: authorization already valid; skipping challenge
2024/11/29 11:13:38 [INFO] [nas.mydomainname.here] acme: Validations succeeded; requesting certificates
2024/11/29 11:13:39 [INFO] [nas.mydomainname.here] Server responded with a certificate.
```
Removed the `acme.json` file and commented out in my server the `LETS_ENCRYPT_CA_SERVER` variable, as it is what I understood from the instructions.

But every time I restart the traefik container, I only see the `acme.json` file being generated, what am I missing from the process? Thank you in advance.

**Comment by AdrienPoupa** [comment]
Hi, I'm not sure I understand. You should have the acme.json file in the letsencrypt folder.

Those variables need to be filled out as described [here](https://github.com/AdrienPoupa/docker-compose-nas?tab=readme-ov-file#traefik-and-ssl-certificates):

```
DNS_CHALLENGE=true
DNS_CHALLENGE_PROVIDER=cloudflare
LETS_ENCRYPT_CA_SERVER="https://acme-v02.api.letsencrypt.org/directory"
LETS_ENCRYPT_EMAIL=
CLOUDFLARE_EMAIL=
CLOUDFLARE_DNS_API_TOKEN=
CLOUDFLARE_ZONE_API_TOKEN=
```

**Comment by castrolem** [comment]
I see, I created the DNS API token with zone dns:edit access, and the Zone API token with zone zone:read access. I'm guessing here's where my mistake lies, but 
<img width="822" alt="image" src="https://github.com/user-attachments/assets/27ba010e-84e5-4901-8d7d-ca76befc187a">

**Comment by AdrienPoupa** [comment]
That looks correct and the same as the documentation: https://go-acme.github.io/lego/dns/cloudflare/#api-tokens

Do you have any Docker logs? You may want to enable debug logging in Traefik.

**Comment by castrolem** [comment]
Ah I see one
```
ERR Unable to obtain ACME certificate for domains error="unable to generate a certificate for the domains [<my private server IP>]: acme: error: 400 :: POST :: https://acme-v02.api.letsencrypt.org/acme/new-order :: urn:ietf:params:acme:error:unsupportedIdentifier :: NewOrder request included invalid non-DNS type identifier: type \"ip\", value \"<my private server IP>\"" ACME CA=https://acme-v02.api.letsencrypt.org/directory acmeCA=https://acme-v02.api.letsencrypt.org/directory domains=["<my private server IP>"] providerName=myresolver.acme routerName=prowlarr@docker rule="(Host(`<my private server IP>`)
```
Basically says that I included a non dns type identifier, and shows my private server IP that I added to the DNS records.

**Comment by AdrienPoupa** [comment]
What does `<my private server IP>` look like? it should be a public domain, not an IP. It should point to your private IP in Tailscale if you're using that.

**Comment by castrolem** [comment]
I'm currently not using Tailscale. Right now it's pointing to the IP of the server in my local network, what should I change in the variables or Cloudflare so that it points to a public one?

**Comment by AdrienPoupa** [comment]
I'm not sure to understand, but basically you should have something like `nas.yourdomain.com` in CloudFlare  pointing to any IP you like, the error message above led me to believe it was trying to generate a certificate for the IP and not the domain name. Make sure `.env`'s `HOSTNAME` is set to your domain, not the IP.

**Comment by castrolem** [comment]
This is how it looks on my CloudFlare dashboard, the black square is my private IP and I do have in my env hostname variable the `nas.mydomain.com` URL. 🙈 

<img width="1287" alt="Screenshot 2024-12-03 at 17 43 16" src="https://github.com/user-attachments/assets/12840bca-f24d-482f-ba41-3ca4014343ac">

**Comment by AdrienPoupa** [comment]
Well, the error definitely means LEGO was trying to generate a certificate for your IP, not your subdomain: `request included invalid non-DNS type identifier: type \"ip\", value \"<my private server IP>\"` https://github.com/cert-manager/cert-manager/issues/6863

Based on Traefik's documentation: https://doc.traefik.io/traefik/https/acme/#domain-definition

> If no [tls.domains](https://doc.traefik.io/traefik/routing/routers/#domains) option is set, then the certificate resolver uses the [router's rule](https://doc.traefik.io/traefik/routing/routers/#rule), by checking the Host() matchers. Please note that [multiple Host() matchers can be used](https://doc.traefik.io/traefik/routing/routers/#certresolver)) for specifying multiple domain names for this router.

So it's going to pick up the domain from here: `traefik.http.routers.sonarr.rule=(Host(`${HOSTNAME}`) && PathPrefix(`/sonarr`))`

Meaning that your `$HOSTNAME` was set to the IP, not the domain. Update it, relaunch the stack with `docker compose up -d` and it should work.

**Comment by castrolem** [comment]
OMG, now I understand, so my issue is that at one point, early in the config, I had my local IP, then I changed the hostname to the one from my CloudFlare configuration, but I was only recreating the Traefik container, instead of doing that I needed to recreate all the containers so all of them could pick up the correct hostname which is what's on the config of each container within the docker compose file. 🤯 amazing.

I tried to see in your website if I could "buy you a coffee" or a similar but didn't see anything, is there any way I could?
